### ISSUE TITLE
Bernstein derivatives yield wrong loss

### ISSUE THREAD

**Issue by LucasKook** [neural-structured-additive-learning/deeptransformation#17]
In `inst/demos/demo-*.R`, I wrote some demos and tests to compare `deeptrafo` and `tram`/`mlt`. For ordinal, count and right-censored survival observations, the two models give virtually the same coefficients and NLL after training. But for continuous responses which use the Bernstein derivatives, wrong NLLs are returned. (Note that `support` and `order` were made comparable.)

```r
set.seed(1234)

# Deps --------------------------------------------------------------------

library(tram)
devtools::load_all("../deepregression/")
devtools::load_all(".")

# Data --------------------------------------------------------------------

data("BostonHousing2", package = "mlbench")

# Unconditional case ------------------------------------------------------

tm <- Colr(cmedv ~ 1, data = BostonHousing2, order = 6,
           support = range(BostonHousing2$cmedv))
cfb <- coef(tm, with_baseline = TRUE)

m <- deeptrafo(cmedv ~ 1, data = BostonHousing2, order = 6,
               optimizer = optimizer_adam(learning_rate = 0.001))
coef(m)

tmp <- get_weights(m$model)
tmp[[2]][] <- 0
tmp[[1]][] <- c(cfb[1], log(exp(diff(cfb)) - 1 + 1e-6))
set_weights(m$model, tmp) # Override weights with tram coefs

logLik(tm); logLik(m)
```
Output:

```
> c(logLik(tm)); logLik(m)
[1] -1774.462
[1] -10334.3
```
